app/Policies/PostPolicy.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rogue\Policies;
+
+use Rogue\Models\Post;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class PostPolicy
+{
+    use HandlesAuthorization;
+
+    /**
+     * Create a new policy instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+    }
+
+    /**
+     * Determine if the given post can be updated by the user.
+     *
+     * @param  \App\User  $user
+     * @return bool
+     */
+    public function update(Post $post)
```

Comment:
I'm having some trouble here - when testing this out I am getting this error: 
```
"message": "Type error: Argument 1 passed to Rogue\\Policies\\PostPolicy::update() must be an instance of Rogue\\Models\\User, instance of DoSomething\\Gateway\\Server\\RemoteUser given, called in /home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php on line 481",
```
does this mean that we're not able to use Permission Policies if getting the `User` not from the User model?

Reply:
oh maybe I have to do something like [this](https://github.com/laravel/framework/commit/70f75255808ffc96275e6f2f356616dd2e163434#diff-961368895033e553787b301c3be0e17a)?

Reply:
@chloealee Do you have a stack trace for that "must be an instance of User" error? It should just be looking for something `Authenticatable`, which `RemoteUser` is.

Reply:
@DFurnes here is the beginning of the stack trace! 
```
    "trace": [
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php",
            "line": 481,
            "function": "update",
            "class": "Rogue\\Policies\\PostPolicy",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php",
            "line": 338,
            "function": "Illuminate\\Auth\\Access\\{closure}",
            "class": "Illuminate\\Auth\\Access\\Gate",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php",
            "line": 313,
            "function": "callAuthCallback",
            "class": "Illuminate\\Auth\\Access\\Gate",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Auth/Access/Gate.php",
            "line": 281,
            "function": "raw",
            "class": "Illuminate\\Auth\\Access\\Gate",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php",
            "line": 23,
            "function": "authorize",
            "class": "Illuminate\\Auth\\Access\\Gate",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/app/Http/Controllers/PostsController.php",
            "line": 146,
            "function": "authorize",
            "class": "Rogue\\Http\\Controllers\\Controller",
            "type": "->"
        },
        {
            "function": "update",
            "class": "Rogue\\Http\\Controllers\\PostsController",
            "type": "->"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Routing/Controller.php",
            "line": 54,
            "function": "call_user_func_array"
        },
        {
            "file": "/home/vagrant/DoSomething.org/rogue/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php",
            "line": 45,
            "function": "callAction",
            "class": "Illuminate\\Routing\\Controller",
            "type": "->"
        },
```

Reply:
Interesting, I just pulled down and I'm getting a different error! I see this:

> Type error: Argument 1 passed to Rogue\Policies\PostPolicy::update() must be an instance of Rogue\Models\Post, instance of DoSomething\Gateway\Server\RemoteUser given

If I update your policy to have the first argument as the `Authenticatable` user, it works!

```php

<?php

use Illuminate\Contracts\Auth\Authenticatable;
// ...

class PostPolicy
{
    // ...

    public function update(Authenticatable $user, Post $post)
    {
        return is_staff_user() || auth()->id() === $post->northstar_id;
    }
}
```

Reply:
You can also then update your conditional to check the passed `$user`'s ID against the post (updated in my code snippet above). It'd probably be a good idea to inline the checks in `is_staff_user()` into this policy then so they can also use the provided user instance.